### PR TITLE
OCPBUGS#10681: Add a new parameter for enabling accelerated networking

### PR DESCRIPTION
**Pull request by xenolinux**
The file `modules/machineset-azure-enabling-accelerated-networking-new-install.adoc` can now be deleted since we have the new parameter to enable accelerated networking

Version(s): 4.11+

Issue: https://issues.redhat.com/browse/OCPBUGS-10681

Link to docs preview: https://58697--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-customizations.html#installation-configuration-parameters-additional-azure_installing-azure-customizations

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -1150,6 +1158,10 @@ at least two zones.
 |The name of the Azure cloud environment that is used to configure the Azure SDK with the appropriate Azure API endpoints. If empty, the default value `AzurePublicCloud` is used.
 |Any valid cloud environment, such as `AzurePublicCloud` or `AzureUSGovernmentCloud`.
 
+|`platform.azure.defaultMachinePlatform.vmNetworkingType`
+|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance.
+|`Accelerated` or `Basic`. The default value is `Accelerated`.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1163]: `The default value is `Accelerated``
This is based on the compute/controlPlane.platform.azure.type setting. If accelerated networking on vm size is supported, installer will enable accelerated networking by default, otherwise, it is Basic.

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -1150,6 +1158,10 @@ at least two zones.
 |The name of the Azure cloud environment that is used to configure the Azure SDK with the appropriate Azure API endpoints. If empty, the default value `AzurePublicCloud` is used.
 |Any valid cloud environment, such as `AzurePublicCloud` or `AzureUSGovernmentCloud`.
 
+|`platform.azure.defaultMachinePlatform.vmNetworkingType`
+|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance.
+|`Accelerated` or `Basic`. If `controlPlane.platform.azure.type` and `compute.platform.azure.type` support `Accelerated` networking, installer by default enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1163]: Actually, user can also set instance type for control plane and worker via `platform.azure.defaultMachinePlatform.type`.
How about rewording as below?
> if instance type of control plane and worker supports `Accelerated` networking, installer by default enables `Accelerated` networking, otherwise the default networking type is `Basic`.

I think same update also needs on `controlPlane.platform.azure.vmNetworkingType` and `compute.platform.azure.vmNetworkingType`

File: modules/installation-configuration-parameters.adoc [hunk 3]
```diff
@@ -1073,6 +1073,10 @@ Additional Azure configuration parameters are described in the following table:
 |Optional. The ID of a disk encryption set in another Azure subscription. This secondary disk encryption set will be used to encrypt compute machines. By default, the installation program will use the disk encryption set from the Azure subscription ID that you provided to the installation program prompts.
 |String, in the format `00000000-0000-0000-0000-000000000000`.
 
+|`compute.platform.azure.vmNetworkingType`
+|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance. If instance type of control plane and worker supports `Accelerated` networking, installer by default enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1077]: ```If instance type of control plane and worker supports `Accelerated` networking```
 -> 
```If instance type of compute machines supports `Accelerated` networking```

File: modules/installation-configuration-parameters.adoc [hunk 4]
```diff
@@ -1101,6 +1105,10 @@ Additional Azure configuration parameters are described in the following table:
 |Enables the use of Azure ultra disks for persistent storage on control plane machines. This requires that your Azure region and zone have ultra disks available.
 |`Enabled`, `Disabled`. The default is `Disabled`.
 
+|`controlPlane.platform.azure.vmNetworkingType`
+|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance. If instance type of control plane and worker supports `Accelerated` networking, installer by default enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1109]: ```If instance type of control plane and worker supports `Accelerated` networking```
-> 
```If instance type of control plane machines supports `Accelerated` networking```

File: modules/installation-configuration-parameters.adoc [hunk 5]
```diff
@@ -1150,6 +1158,10 @@ at least two zones.
 |The name of the Azure cloud environment that is used to configure the Azure SDK with the appropriate Azure API endpoints. If empty, the default value `AzurePublicCloud` is used.
 |Any valid cloud environment, such as `AzurePublicCloud` or `AzureUSGovernmentCloud`.
 
+|`platform.azure.defaultMachinePlatform.vmNetworkingType`
+|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance.
+|`Accelerated` or `Basic`. If instance type of control plane and worker supports `Accelerated` networking, installer by default enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1163]: ```If instance type of control plane and worker supports `Accelerated` networking```
->
```If instance type of control plane machines and compute machines supports `Accelerated` networking```

**Comment by jinyunma** [comment]
@xenolinux thanks for addressing, general change looks good to me, except a few minor suggestion.

**Comment by xenolinux** [comment]
@jinyunma Addressed the suggested change. Can you PTAL again?
Thank you

File: modules/installation-configuration-parameters.adoc [hunk 6]
```diff
@@ -1150,6 +1158,10 @@ at least two zones.
 |The name of the Azure cloud environment that is used to configure the Azure SDK with the appropriate Azure API endpoints. If empty, the default value `AzurePublicCloud` is used.
 |Any valid cloud environment, such as `AzurePublicCloud` or `AzureUSGovernmentCloud`.
 
+|`platform.azure.defaultMachinePlatform.vmNetworkingType`
+|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance.
+|`Accelerated` or `Basic`. If instance type of control plane and worker machines support `Accelerated` networking, installer by default enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1163]: If instance type of control plane and worker machines
-> 
`If instance type of control plane machines and compute machines`

Comment (xenolinux) [reply modules/installation-configuration-parameters.adoc]: Addressed

File: modules/installation-configuration-parameters.adoc [hunk 7]
```diff
@@ -1073,6 +1073,10 @@ Additional Azure configuration parameters are described in the following table:
 |Optional. The ID of a disk encryption set in another Azure subscription. This secondary disk encryption set will be used to encrypt compute machines. By default, the installation program will use the disk encryption set from the Azure subscription ID that you provided to the installation program prompts.
 |String, in the format `00000000-0000-0000-0000-000000000000`.
 
+|`compute.platform.azure.vmNetworkingType`
+|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance. If instance type of compute machines support `Accelerated` networking, installer by default enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```
Comment (snarayan-redhat) [inline modules/installation-configuration-parameters.adoc:1077]:
```suggestion
|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance. If instance type of compute machines support `Accelerated` networking, by default, the installer enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```

File: modules/installation-configuration-parameters.adoc [hunk 8]
```diff
@@ -1101,6 +1105,10 @@ Additional Azure configuration parameters are described in the following table:
 |Enables the use of Azure ultra disks for persistent storage on control plane machines. This requires that your Azure region and zone have ultra disks available.
 |`Enabled`, `Disabled`. The default is `Disabled`.
 
+|`controlPlane.platform.azure.vmNetworkingType`
+|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance. If instance type of control plane machines support `Accelerated` networking, installer by default enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```
Comment (snarayan-redhat) [inline modules/installation-configuration-parameters.adoc:1109]:
```suggestion
|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance. If instance type of control plane machines support `Accelerated` networking, by default, the installer enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```

File: modules/installation-configuration-parameters.adoc [hunk 9]
```diff
@@ -1150,6 +1158,10 @@ at least two zones.
 |The name of the Azure cloud environment that is used to configure the Azure SDK with the appropriate Azure API endpoints. If empty, the default value `AzurePublicCloud` is used.
 |Any valid cloud environment, such as `AzurePublicCloud` or `AzureUSGovernmentCloud`.
 
+|`platform.azure.defaultMachinePlatform.vmNetworkingType`
+|Enables accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, improving its networking performance.
+|`Accelerated` or `Basic`. If instance type of control plane and compute machines support `Accelerated` networking, installer by default enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```
Comment (snarayan-redhat) [inline modules/installation-configuration-parameters.adoc:1163]:
```suggestion
|`Accelerated` or `Basic`. If instance type of control plane and compute machines support `Accelerated` networking, by default, the installer enables `Accelerated` networking, otherwise the default networking type is `Basic`.
```

File: modules/installation-configuration-parameters.adoc (hunk 7 above)
Comment (snarayan-redhat) [reply modules/installation-configuration-parameters.adoc:1077]: Also the `If instance type of compute machines support...` statement is mentioned in `Values` column for the other 2 fields. can you make this consistent?

File: modules/installation-configuration-parameters.adoc (hunk 8 above)
Comment (snarayan-redhat) [reply modules/installation-configuration-parameters.adoc:1109]: Also the `If instance type of compute machines support...` statement is mentioned in `Values` column for the other 2 fields. can you make this consistent?

**Review by snarayan-redhat — COMMENTED** [review]
Looks good overall. Left few comments.
